### PR TITLE
Plugin now stays in memory until IDA terminates completely. Closes #8.

### LINKED ISSUE #8: IDA crashing on termination.
It seems that every time I try and close IDA, it crashes.

It is not critical, as it crashes after saving the database, but it is still a bug. As it does no noticeable harm, I only noticed it after setting Windbg to be the postmortem debugger. But once I set it, I now see IDA crashing due to an access violation on every termination.

This has been reproduced on multiple computers.

Will add more information as soon as my debugging yields any.

### PR DESCRIPTION
**Pull request by tmr232**
The patch makes the plugin module stay in memory until IDA terminates, or as MSDN puts it:

> The module stays loaded until the process is terminated, no matter how many times FreeLibrary is called.

From [GetModuleHandleEx](https://msdn.microsoft.com/en-us/library/windows/desktop/ms683200%28v=vs.85%29.aspx)

This solves the issue with the hook release, closes #8.

### CODE REVIEW

File: src/ida_ipython.cpp [hunk 1]
```diff
@@ -3,6 +3,8 @@
 #include "pro.h"
 #include "ida.hpp"
 
+#include "persist.c"
```
Comment (james91b) [inline src/ida_ipython.cpp]: I think this should be including the header file, not the c file. We could either just write persist in C++ (just change the file name to .cpp) and include the header file, or wrap the include in an extern "C"

File: src/ida_ipython.cpp [hunk 2]
```diff
@@ -56,6 +59,15 @@ int idaapi init(void)
         }
         return PLUGIN_SKIP;
     }
+
+
+    /* Try and make the module persist in memory until termination. Failing to do so can cause IDA to crash
+       when it terminates the plugin. */
+    success = persist();
+    if (0 != success) {
+        return PLUGIN_SKIP;
```
Comment (james91b) [inline src/ida_ipython.cpp:69]: Maybe we should add in a warning here so we know what went wrong?

File: src/ida_ipython.cpp (hunk 1 above)
Comment (tmr232) [inline src/ida_ipython.cpp]: You are quite right. I seem to be a bit out of practice in C.

**Comment by james91b** [comment]
Thank you. Looks good, great work! :+1:

**Pull request merged**